src/System.IO.Pipelines/tests/PipeReaderWriterFacts.cs

```diff
@@ -149,15 +149,26 @@ public void Dispose()
         }
 
         [Fact]
-        public async Task CompleteReaderThrowsIfReadInProgress()
+        public async Task CompleteReaderAfterFlushWithoutAdvancing()
+        {
+            await _pipe.Writer.FlushAsync();
+            ReadResult result = await _pipe.Reader.ReadAsync();
+            ReadOnlySequence<byte> buffer = result.Buffer;
+
+            _pipe.Reader.Complete();
```

Comment:
Does the pipe get reset correctly after this?

Reply:
What do you mean? Are you asking if Reset works correctly if I complete both sides where the reader never called Advance? I can add a test for that.

Reply:
Yes exactly that  scenario